dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.rpm.Rpm.xml

```diff
@@ -212,6 +212,23 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
         <arg name="options" type="a{sv}" direction="in" />
     </method>
 
+    <!--
+        system_upgrade:
+        @options: an array of key/value pairs to modify system_upgrade behavior
+
+        Prepare a transaction for upgrade to the new distribution release. The prepared transaction should be executed during reboot (see `offline` option of the `do_transaction` method of the `Goal` interface).
+
+        Following @options are supported:
+
+            - mode: string (one of "distrosync", "upgrade", default is "distrosync")
+                By default the system_upgrade behaves like `dnf distro-sync`, always installing packages from the new release, even if they are older then the currently installed version. If set to "upgrade", packages from the new release are not installed if they are older then what is currently installed (behave like `dnf upgrade`).
+
+        Unknown options are ignored.
```

Comment:
how to I set the release version to be upgraded to? It seems to be "detected" somehow, but one can upgrade both to N+1 and N+2. An explicit "upgrade to `releasever` X" would be better.

The python example suggests the `releasever` is set within the `open_session`. Could there be added an argument, which will override that value if set, here as well, please?

Reply:
Releasever should be set using the `open_session()` option. It needs to be set before the repositories metadata are loaded and setting it here can be too late. I'll add this information to the method description.